tests/rl/test_hooks.py

```diff
@@ -0,0 +1,472 @@
+)
+
+
+class TestHookContext(unittest.TestCase):
+    """Test HookContext functionality."""
+
+    def setUp(self):
+        """Set up test fixtures."""
+        self.mock_worker = MagicMock()
+        self.mock_curriculum_actor = MagicMock()
+        self.rng = jrandom.PRNGKey(42)
+
+    def _create_context(self, **kwargs):
+        """Helper to create HookContext with required fields."""
+        defaults = {
+            "worker": self.mock_worker,
+            "step": 100,
+            "rng": self.rng,
+            "curriculum_actor": self.mock_curriculum_actor,
+        }
+        defaults.update(kwargs)
+        return HookContext(**defaults)
+
+    def test_hook_context_creation(self):
+        """Test creating a hook context."""
+        context = self._create_context(lesson_id="test_lesson", metadata={"key": "value"})
+
+        self.assertEqual(context.worker, self.mock_worker)
+        self.assertEqual(context.step, 100)
+        self.assertEqual(context.lesson_id, "test_lesson")
+        self.assertEqual(context.metadata["key"], "value")
+
+    def test_split_rng(self):
+        """Test RNG splitting in context."""
+        context = self._create_context()
+        rng1, rng2 = context.split_rng()
+
+        # Check that we get two different RNG keys
+        self.assertFalse(jnp.array_equal(rng1, rng2))
+        self.assertFalse(jnp.array_equal(self.rng, rng1))
+        self.assertFalse(jnp.array_equal(self.rng, rng2))
+
+
+class TestPeriodicHook(unittest.TestCase):
+    """Test PeriodicHook functionality."""
+
+    def setUp(self):
+        """Set up test fixtures."""
+        self.mock_worker = MagicMock()
+        self.mock_curriculum_actor = MagicMock()
+        self.rng = jrandom.PRNGKey(42)
+
+    def _create_context(self, **kwargs):
+        """Helper to create HookContext with required fields."""
+        defaults = {
+            "worker": self.mock_worker,
+            "step": 100,
+            "rng": self.rng,
+            "curriculum_actor": self.mock_curriculum_actor,
+        }
+        defaults.update(kwargs)
+        return HookContext(**defaults)
+
+    def test_periodic_hook_frequency(self):
+        """Test that periodic hooks run at the correct frequency."""
+
+        class TestHook(PeriodicHook):
+            def run(self, context):
+                return {"ran": True}
+
+        hook = TestHook(frequency=10, start_step=0)
+
+        # Should not run at step 5
+        context = self._create_context(step=5)
+        self.assertFalse(hook.should_run(context))
+
+        # Should run at step 10
+        context = self._create_context(step=10)
+        self.assertTrue(hook.should_run(context))
+
+        # Should not run at step 15
+        context = self._create_context(step=15)
+        self.assertFalse(hook.should_run(context))
+
+        # Should run at step 20
+        context = self._create_context(step=20)
+        self.assertTrue(hook.should_run(context))
+
+    def test_periodic_hook_start_step(self):
```

Comment:
you can define TestHook at the top level and reuse, or merge these tests to share.